src/NuGet.Core/NuGet.Protocol.VisualStudio/ExtensibleSourceRepositoryProvider.cs

```diff
@@ -84,17 +87,19 @@ public Configuration.IPackageSourceProvider PackageSourceProvider
             get { return _packageSourceProvider; }
         }
 
-        private void Init()
+        private void ResetRepositories()
         {
-            _repositories.Clear();
+            var repositories = new List<SourceRepository>();
             foreach (var source in _packageSourceProvider.LoadPackageSources())
             {
                 if (source.IsEnabled)
                 {
                     var sourceRepo = new SourceRepository(source, _resourceProviders);
-                    _repositories.Add(sourceRepo);
+                    repositories.Add(sourceRepo);
                 }
             }
+
+            Interlocked.CompareExchange(ref _repositories, repositories, null);
```

Comment:
Is this interlocked needed? I only see it used in one place, I would expect this to either just assign it normally, or have two usages.